UnityProject/Assets/Scripts/Robotics/Bots/InteractableBot.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Collections;
+using UnityEngine;
+using Mirror;
+using Systems.MobAIs;
+
+namespace Robotics
+{
+	public class InteractableBot : NetworkBehaviour, ICheckedInteractable<HandApply>
+	{
+
+		[SerializeField] private SpriteHandler SPRITE_HANDLER;
+		[SerializeField] private Sprite EMAGGED_SPRITE;
+
+		private MobExplore mobController;
+		public MobExplore MobController
+		{
+			get
+			{
+				if (!mobController)
+				{
+					mobController = GetComponent<MobExplore>();
+				}
+				return mobController;
+			}
+		}
+
+		public bool WillInteract(HandApply interaction, NetworkSide side)
+		{
+			if (!DefaultWillInteract.Default(interaction, side)) return false;
+			if (interaction.TargetObject != gameObject) return false;
+			if (interaction.HandObject != null && interaction.Intent == Intent.Harm) return false;
+
+			return MobController != null;
+		}
+		public void ServerPerformInteraction(HandApply interaction)
+		{
+			if (Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.Emag)
+				&& interaction.HandObject.TryGetComponent<Emag>(out var emag)
+				&& emag.EmagHasCharges())
+			{
+				PerformEmag(emag, interaction);
+			}
+		}
+		public void PerformEmag(Emag emag, HandApply interaction)
+		{
+			if (MobController == null) return;
+			MobController.IsEmagged = true;
+			if (EMAGGED_SPRITE != null && SPRITE_HANDLER != null) SPRITE_HANDLER.SetSprite(EMAGGED_SPRITE);
```

Comment:
does this work? I remember SpriteHandler used indexes instead of sprites.

Reply:
I've run it and it works fine

Reply:
aight, good